zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/zos/security/service/AccessControlService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.commons.zos.security.service;
+
+import org.zowe.commons.zos.security.platform.PlatformAccessControl;
+import org.zowe.commons.zos.security.platform.PlatformAckErrno;
+import org.zowe.commons.zos.security.platform.PlatformErrno2;
+import org.zowe.commons.zos.security.platform.PlatformReturned;
+import org.zowe.commons.zos.security.platform.PlatformAccessControl.AccessLevel;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public abstract class AccessControlService implements PlatformSecurityService {
+    protected PlatformAccessControl platformAccessControl = null;
+
+    protected PlatformAccessControl getPlatformAccessControl() {
+        return platformAccessControl;
+    }
+
+    @Override
+    public boolean checkPermission(String userid, String resourceClass, String resourceName, AccessLevel accessLevel,
+            boolean resourceHasToExist) {
+        PlatformReturned returned = getPlatformAccessControl().checkPermission(userid, resourceClass, resourceName,
+                accessLevel.getValue());
+        return evaluatePlatformReturned(returned, resourceHasToExist);
+    }
+
+    private boolean evaluatePlatformReturned(PlatformReturned returned, boolean resourceHasToExist) {
+        if (returned == null) {
+            return true;
+        } else {
+            String message;
+            PlatformAckErrno errno = PlatformAckErrno.valueOfErrno(returned.errno);
+            PlatformErrno2 errno2 = PlatformErrno2.valueOfErrno(returned.errno2);
+            if ((errno == null) || (errno2 == null)) {
+                message = "Unknown access control error";
+                log.error("Platform access control failed: {}", returned);
+            } else {
+                message = "Platform access control failed: " + errno2.explanation;
+                if (!resourceHasToExist && (errno2 == PlatformErrno2.JRSAFResourceUndefined)) {
+                    return true;
+                } else if (errno2 == PlatformErrno2.JRNoResourceAccess) {
+                    return false;
+                } else if (errno2 == PlatformErrno2.JRSAFResourceUndefined) {
+                    return false;
+                }
+                log.error("Platform access control failed: {} {} {}", errno.name, errno2.name, errno2.explanation,
```

Comment:
perhaps need another `{}` for substitution here?

Reply:
yes, thank you